main.nf

```diff
@@ -459,6 +501,91 @@ process fastp {
     """
 }
 
+/*
+ * Remove host read contamination
+ */
+(trimmed_reads, ch_trimmed_reads_remove_host) = trimmed_reads.into(2)
+
+process host_bowtie2index {
+    tag "${genome}"
+
+    input:
+    file(genome) from ch_host_fasta
+
+    output:
+    file("bt2_index_base*") into ch_host_bowtie2index
+
+    when: params.host_fasta
+
+    script:
+    """
+    bowtie2-build --threads "${task.cpus}" "${genome}" "bt2_index_base"
+    """
+}
+
+process remove_host {
+    tag "${name}"
+
+    publishDir "${params.outdir}/QC_shortreads/remove_host/", mode: 'copy',
+        saveAs: {filename ->
+                    if (filename.indexOf("read_ids.txt") > 0) "$filename"
+                    else null
+                }
+
+    input:
+    set val(name), file(reads) from ch_trimmed_reads_remove_host
+    file(index) from ch_host_bowtie2index
+
+    output:
+    set val(name), file("${name}_host_unmapped*.fastq.gz") into ch_trimmed_reads_host_removed
+    file("${name}.bowtie2.log") into ch_host_removed_log
+    file("${name}_host_mapped*.read_ids.txt") into ch_mapped_read_ids
+
+    when: params.host_fasta || params.host_genome
+
+    script:
+    def sensitivity = params.host_removal_verysensitive ? "--very-sensitive" : "--sensitive"
+    if ( !params.single_end ) {
+        """
+        bowtie2 -p "${task.cpus}" \
+                -x ${index[0].getSimpleName()} \
+                -1 "${reads[0]}" -2 "${reads[1]}" \
+                $sensitivity \
+                --un-conc-gz ${name}_host_unmapped_%.fastq.gz \
+                --al-conc-gz ${name}_host_mapped_%.fastq.gz \
+                1> /dev/null \
+                2> ${name}.bowtie2.log
+
+        zcat ${reads[0]} | echo "Read pairs before removal: \$((`wc -l`/4))" >>${name}_remove_host.log
+        zcat ${name}_host_unmapped_1.fastq.gz | echo "Read pairs after removal: \$((`wc -l`/4))" >>${name}_remove_host.log
```

Comment:
Is this just for debugging? I can't see where `${name}_remove_host.log` is used elsewhere.

Might be a bit of an expensive operation if so, `zcat` on a big FastQ file can take quite a while. And I guess we get this info from your bowtie logs MultiQC module anyway?

Reply:
True, I originally copied it from the `remove_phix` process, but it can be removed here.